### ISSUE TITLE
Publish to GitHub uses master instead of default

### ISSUE THREAD

**Issue by BillRaymond** [microsoft/vscode#105025]
Issue Type: <b>Bug</b>

1. On the local computer, create a default branch that is not master, such as main by typing the following: `git config --global init.defaultBranch main`
2. Create a new VSC folder and run the command `Git: Initialize repository' 
3. Create and commit a readme.md file
4. Publish the repository using `Publish to GitHub'. In my case, I chose GitHub Public
5. Go to the GitHub website and open the repository to find there is no code because it is using the `master` branch. Oddly, you cannot select a new default branch with this use case (at least right now?)

Note: I tried the `Publish branch` option, but received errors informing me I had to pull, which resulted in additional errors. Ultimately it seems to have worked(?), but it does not feel like a good workflow. I think most people like myself will want to use the Publish to GitHub option and have it automatically sort out the fact you are pushing the configured default branch.

VS Code version: Code 1.48.0 (db40434f562994116e5b21c24015a2e40b2504e6, 2020-08-13T08:03:27.059Z)
OS version: Darwin x64 19.6.0

<details><summary>Extensions (4)</summary>

Extension|Author (truncated)|Version
---|---|---
vscode-liquid-snippets|kil|2.0.0
language-liquid|nei|0.1.1
live-sass|rit|3.0.0
LiveServer|rit|5.6.1

</details>

**Comment by nrayburn-tech** [comment]
It looks like it always uses `master` when using `Publish to Github`.  https://github.com/microsoft/vscode/blob/b58f4549b2919fa3a5dc52df83f4ba60780c625f/extensions/github/src/publish.ts#L178

I am not entirely sure how to get the current branch to make a PR though.

**Comment by AndreaRe9** [comment]
In order to get the current branch, I think that we could use the method below to get the global configuration and then, based on the `defaultBranch` parameter, we could run the `repository.push()` method. Does it seem feasible to you, @joaomoreno?

https://github.com/microsoft/vscode/blob/b58f4549b2919fa3a5dc52df83f4ba60780c625f/extensions/github/src/typings/git.d.ts#L152

**Comment by joaomoreno** [comment]
No, I would actually try to get the name of the current branch from the repo itself with `git symbolic-ref --short HEAD`.

**Comment by BillRaymond** [comment]
I do not know how helpful this is, but GitHub links to this page as their official updates on how they are defaulting to `main`:
[Renaming the default branch from master](https://github.com/github/renaming)

There is useful guidance on there, but it is not directed at developers who are creating tooling like VSC. Of course, GitHub is also enabling any default branch, which is why my concern with VSC is not just supporting `main`, but any primary branch name. Hope this helps.

**Comment by nrayburn-tech** [comment]
The PR I submitted will use whatever branch name is configured.  Whether it is `master`, `main`, `something-else`.

**Comment by BillRaymond** [comment]
Thanks!

**Comment by rzhao271** [comment]
I initialized a git repo with branch name "whatever".
It seems that 'Publish to GitHub' creates the repo, but it doesn't pick up the branch name?
I tried cloning the created repo and it seems to be completely empty.

![workspace-made-but-empty](https://user-images.githubusercontent.com/7199958/94840703-aff24000-03cd-11eb-8bf2-3d43f299993b.PNG)

**Comment by joaomoreno** [comment]
@rzhao271 Was there an error? How did you initialized the git repo with that branch name?

This seems to work just fine for me:

```
git init
git checkout -b whatever
```

Then create one commit. Then publish to Github.

![image](https://user-images.githubusercontent.com/22350/94900175-d16a2f00-0494-11eb-8ae4-f9cd2c05afc6.png)

**Comment by nrayburn-tech** [comment]
I think if you commit and publish without any files, you get the empty repo without a branch.

**Comment by rzhao271** [comment]
I followed the steps in the original post, so I did not run `git init` or `git checkout -b whatever` directly in a terminal. I also published with the readme.md file selected.
Same result (empty repo after publish) after updating to today's insiders.

**Comment by eamodio** [comment]
I'm seeing a lot going wrong here and will open a few new issues. But if I go down the "happy path" of not having a git repo at all, it now publishes to my local default branch, rather than the default branch I have setup on GitHub. I would expect it to use the GitHub defaults as I'm publishing there.

**Comment by BillRaymond** [comment]
I am on vacation this week, so will test when I return

**Comment by joaomoreno** [comment]
> rather than the default branch I have setup on GitHub. I would expect it to use the GitHub defaults as I'm publishing there.

I would call this a separate issue, thanks for filling that as a separate one!

This feature simply respects whatever branch you have created in your local repo.

**Comment by joaomoreno** [comment]
\closedWith 1562816430ff36deb6bd1d220c4a5d44755a0abe

**Comment by BillRaymond** [comment]
Thank you for looking into this, and sorry I could not test last week as I was on vacation. 

I downloaded the Insider's build (1.50.0-insider) and initialized two different local repos with different names for the mainline branch (one named `main` and one named `billraymond`. Each repo includes a _readme.md_ file that I commit before publishing. Each of those local repos published to GitHub without error. Opening the repos in GitHub, I see they are using the correct branch names I originally instantiated locally.

Going one step further, I cloned each repo from GitHub to my local computer. The proper mainline branches pulled down. I was able to commit new files and sync without error.

/verified
